extensions/wikia/CreatePage/CreatePage.php

```diff
@@ -122,8 +122,13 @@ function wfCreatePageLoadPreformattedContent( $editpage ) {
 			} else {
 				$editpage->textbox1 = wfMsgForContentNoTrans( 'newpagelayout' );
 			}
-		} else if ( $msg = $wgRequest->getVal( 'useMessage' ) ) {
-			$editpage->textbox1 = wfMsgForContentNoTrans( $msg );
+		} else if ( $msgKey = $wgRequest->getVal( 'useMessage' ) ) {
+			$msg = wfMessage( $msgKey );
+			$msgParams = $wgRequest->getArray( 'messageParams' );
+			if ( $msgParams !== null ) {
+				$msg = $msg->params( $msgParams );
+			}
+			$editpage->textbox1 = $msg->escaped();
```

Comment:
`textbox1` should be escaped on output (obviously double check ;)) as it is run through `Html::textarea` so this would result in double escaping. Also, since it should replicate `wfMsgForContentNoTrans`, this is currently both missing the `->inContentLanguage()` (so it will be in the user's language rather than the wikia's) and it's doing the Trans step as part of escaped. So, the output format should be `->plain()` so it is "NoTrans" (unless we want the transformations to happen in the wikitext now?) with any extra escaping being done with `htmlspecialchars` or whatnot.